File: app/javascript/channels/mission_channel.js
import consumer from "./consumer"

const missionId = document.body.dataset.missionId;
if (missionId) {
  consumer.subscriptions.create(
    { channel: "MissionChannel", mission_id: missionId },
    {
      connected() {
        console.log("Connected to MissionChannel for mission", missionId);
      },
      disconnected() {
        console.log("Disconnected from MissionChannel");
      },
      received(data) {
        console.log("Received data:", data);
      }
    }
  );
} else {
  console.warn("mission_id is null, not subscribing to MissionChannel");
}

File: app/channels/mission_channel.rb
# app/channels/mission_channel.rb
class MissionChannel < ApplicationCable::Channel
  def subscribed
    @mission = Mission.find(params[:mission_id])  # Find the mission the user is subscribing to
    Rails.logger.info "---------------------------"
    Rails.logger.info "Subscribing to MissionChannel with mission_id: #{params[:mission_id]}"
    Rails.logger.info "---------------------------"
    stream_for @mission  # Start streaming updates for this specific mission
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
